tests/titan/titan.go

```diff
@@ -0,0 +1,160 @@
+package titan
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	pd "github.com/pingcap/pd/client"
+	"github.com/tikv/client-go/config"
+	"github.com/tikv/client-go/rawkv"
+
+	"github.com/ngaut/log"
+
+	"github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/core"
+	"github.com/pingcap/tipocket/util"
+)
+
+// CaseCreator is a creator of a read-stress test
+type CaseCreator struct {
+}
+
+// Create creates a read-stress test client
+func (c CaseCreator) Create(node types.ClientNode) core.Client {
+	return &titanClient{}
+}
+
+type titanClient struct {
+	cli *rawkv.Client
+	pd  pd.Client
+}
+
+// SetUp implements the core.Client interface.
+func (c *titanClient) SetUp(ctx context.Context, nodes []types.ClientNode, idx int) error {
+	log.Infof("setup client %v start", idx)
+
+	clusterName := nodes[0].ClusterName
+	ns := nodes[0].Namespace
+	pdAddrs := []string{fmt.Sprintf("%s-pd.%s.svc:2379", clusterName, ns)}
+	if len(pdAddrs) == 0 {
+		return errors.New("No pd node found")
+	}
+
+	cfg := config.Default()
+	cfg.Raw.MaxScanLimit = 40000
+	var err error
+	c.cli, err = rawkv.NewClient(ctx, pdAddrs, cfg)
+	if err != nil {
+		log.Fatalf("create tikv client error: %v", err)
+	}
+
+	log.Infof("setup client %v end", idx)
+
+	return nil
+}
+
+// TearDown implements the core.Client interface.
+func (c *titanClient) TearDown(ctx context.Context, nodes []types.ClientNode, idx int) error {
+	return nil
+}
+
+func (c *titanClient) Invoke(ctx context.Context, node types.ClientNode, r interface{}) core.UnknownResponse {
+	panic("implement me")
+}
+
+func (c *titanClient) NextRequest() interface{} {
+	panic("implement me")
+}
+
+func (c *titanClient) DumpState(ctx context.Context) (interface{}, error) {
+	panic("implement me")
+}
+
+func (c *titanClient) Start(ctx context.Context, cfg interface{}, clientNodes []types.ClientNode) error {
+	for {
+		for i := 1; i <= 100; i++ {
+			rnd := rand.New(rand.NewSource(time.Now().Unix()))
+			keyHashMap := make(map[string]uint32)
+			hashs := make(map[uint32]struct{})
+
+			for j := 0; j <= 40000; j++ {
+				key := fmt.Sprintf("v%3dj%5d", i, j)
```

Comment:
`"v%3dj%5d"` will produce string like `v  1j     1`, which is padded with whitespace, I think string padding with zero is more readable.